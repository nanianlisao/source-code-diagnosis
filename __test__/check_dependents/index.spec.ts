import { expect, test } from "vitest";
import { fileURLToPath } from "node:url";
import path, { dirname } from "node:path";
import { checkDependents, type Graphics } from "../../index.js";

const __filename = fileURLToPath(import.meta.url);

function normalizePaths(graphics: Graphics): string {
	const dictionaries = graphics.dictionaries;

	const graph = graphics.graph
		.map((x) => `${dictionaries[x.source]} -> ${dictionaries[x.target]}`)
		.sort()
		.join(" :: ");

	return graph;
}

test("Get which files depend on the specified file", () => {
	const cwd = path.resolve(dirname(__filename), "features", "normal");

	const response = checkDependents("utils.js", {
		cwd,
	});

	const normalizedPaths = normalizePaths(response);

	expect(normalizedPaths).toMatchSnapshot();
});

test("Get which files depend on the specified file with alias", () => {
	const cwd = path.resolve(dirname(__filename), "features", "alias");
	const response = checkDependents("utils.js", {
		alias: {
			"@": [cwd],
		},
		cwd,
	});
	const normalizedPaths = normalizePaths(response);

	expect(normalizedPaths).toMatchSnapshot();
});
