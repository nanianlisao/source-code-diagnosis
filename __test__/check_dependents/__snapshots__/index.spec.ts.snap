// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Get which files depend on the specified file 1`] = `
[
  [
    {
      "astNode": {
        "loc": {
          "end": {
            "col": 29,
            "line": 1,
          },
          "start": {
            "col": 1,
            "line": 1,
          },
        },
        "span": {
          "end": 28,
          "start": 0,
        },
      },
      "source": "/a.js",
      "target": "/utils.js",
    },
  ],
  [
    {
      "astNode": {
        "loc": {
          "end": {
            "col": 29,
            "line": 1,
          },
          "start": {
            "col": 1,
            "line": 1,
          },
        },
        "span": {
          "end": 28,
          "start": 0,
        },
      },
      "source": "/a.js",
      "target": "/utils.js",
    },
    {
      "astNode": {
        "loc": {
          "end": {
            "col": 25,
            "line": 1,
          },
          "start": {
            "col": 1,
            "line": 1,
          },
        },
        "span": {
          "end": 24,
          "start": 0,
        },
      },
      "source": "/c.js",
      "target": "/a.js",
    },
  ],
  [
    {
      "astNode": {
        "loc": {
          "end": {
            "col": 29,
            "line": 1,
          },
          "start": {
            "col": 1,
            "line": 1,
          },
        },
        "span": {
          "end": 28,
          "start": 0,
        },
      },
      "source": "/b.js",
      "target": "/utils.js",
    },
  ],
  [
    {
      "astNode": {
        "loc": {
          "end": {
            "col": 29,
            "line": 1,
          },
          "start": {
            "col": 1,
            "line": 1,
          },
        },
        "span": {
          "end": 28,
          "start": 0,
        },
      },
      "source": "/b.js",
      "target": "/utils.js",
    },
    {
      "astNode": {
        "loc": {
          "end": {
            "col": 25,
            "line": 2,
          },
          "start": {
            "col": 1,
            "line": 2,
          },
        },
        "span": {
          "end": 49,
          "start": 25,
        },
      },
      "source": "/c.js",
      "target": "/b.js",
    },
  ],
]
`;

exports[`Get which files depend on the specified file with alias 1`] = `
[
  [
    {
      "astNode": {
        "loc": {
          "end": {
            "col": 29,
            "line": 1,
          },
          "start": {
            "col": 1,
            "line": 1,
          },
        },
        "span": {
          "end": 28,
          "start": 0,
        },
      },
      "source": "/a.js",
      "target": "/utils.js",
    },
  ],
  [
    {
      "astNode": {
        "loc": {
          "end": {
            "col": 29,
            "line": 1,
          },
          "start": {
            "col": 1,
            "line": 1,
          },
        },
        "span": {
          "end": 28,
          "start": 0,
        },
      },
      "source": "/a.js",
      "target": "/utils.js",
    },
    {
      "astNode": {
        "loc": {
          "end": {
            "col": 25,
            "line": 1,
          },
          "start": {
            "col": 1,
            "line": 1,
          },
        },
        "span": {
          "end": 24,
          "start": 0,
        },
      },
      "source": "/c.js",
      "target": "/a.js",
    },
  ],
  [
    {
      "astNode": {
        "loc": {
          "end": {
            "col": 29,
            "line": 1,
          },
          "start": {
            "col": 1,
            "line": 1,
          },
        },
        "span": {
          "end": 28,
          "start": 0,
        },
      },
      "source": "/b.js",
      "target": "/utils.js",
    },
  ],
  [
    {
      "astNode": {
        "loc": {
          "end": {
            "col": 29,
            "line": 1,
          },
          "start": {
            "col": 1,
            "line": 1,
          },
        },
        "span": {
          "end": 28,
          "start": 0,
        },
      },
      "source": "/b.js",
      "target": "/utils.js",
    },
    {
      "astNode": {
        "loc": {
          "end": {
            "col": 25,
            "line": 2,
          },
          "start": {
            "col": 1,
            "line": 2,
          },
        },
        "span": {
          "end": 49,
          "start": 25,
        },
      },
      "source": "/c.js",
      "target": "/b.js",
    },
  ],
]
`;
